pkg/koordlet/koordlet.go

```diff
@@ -152,9 +153,11 @@ func (d *daemon) Run(stopCh <-chan struct{}) {
 		}
 	}()
 
-	// TODO add HasSync function for collector
-	klog.Infof("waiting 10 seconds for collector synced before start reporter")
-	time.Sleep(10 * time.Second)
+	// wait for collector sync
+	if !cache.WaitForCacheSync(stopCh, d.collector.HasSynced) {
```

Comment:
Theoretically, we can run collector in a normal way instead of using goroutine. Then we don't need a HasSynced function for collector. 
@zwzhang0107 WDYT?